simplify-basedomain-usage/README.md

```diff
@@ -0,0 +1,186 @@
+    app.kubernetes.io/version: 0.26.0
+    application.giantswarm.io/team: hydra
+    cluster-apps-operator.giantswarm.io/watching: ""
+    cluster.x-k8s.io/cluster-name: golem
+    giantswarm.io/cluster: golem
+    giantswarm.io/organization: giantswarm
+    helm.sh/chart: cluster-aws-0.26.0
+  name: golem
+  namespace: org-giantswarm
+```
+
+The `baseDomain` would still be used by the applications that we deploy to management clusters that don't reconcile clusters, because these applications need the DNS zone of the management cluster.
+This variable could potentially be renamed to something like `managementClusterDNSZone` in the future to make clear its purpose, but it's not in the scope of this RFC to discuss this.
+
+### Advantages
+
+On top of reducing the usage of the global `baseDomain` value, there are other advantages when using this approach.
+
+#### Custom Resources triggering changes rather than controllers
+
+Moving configuration from operators to custom resources is always preferable. Starting by the obvious, the cluster configuration is right there explicitly in the `Cluster` Custom resource rather than hidden in controllers configuration. 
+Furthermore, if the value is part of the `Cluster` CR, it's the `Cluster` the one driving and triggering changes on cloud resources, allowing clusters to change or update their values independently of each other.
+This is a better scenario than our current approach having the controller receive this configuration as a parameter, because changing the parameter would trigger changes in cloud resources on all clusters.
+
+#### Reduce coupling
+
+Controllers fetching the `baseDomain` value from the `cluster-values` configmap depend on the `cluster-apps-operator` or `cluster-operator` deployed and creating the configmap.
+Using the value from the annotation instead of depending on the configmap reduces coupling between our controllers.
+
+#### Potential to use different DNS hosted zones on the same Management Cluster
+
+With our current approach, we configure the `baseDomain` at the installation level through the `config` repository, making the `baseDomain` shared for all clusters in a single installation.
+If we move the value to be part of the `Cluster` CR, different clusters could potentially use different hosted zones, even on the same Management Cluster.
+
+#### Migrate workload clusters to different management cluster
+
+One side effect from this is that by decoupling the `baseDomain` from the Management Cluster would also help moving clusters between installations, if that would be desired.
+
+
+### Downsides
+
+#### Migration
+
+For the migration to this approach, we would need to annotate all `Cluster` Custom Resources to have the annotation.
+We also would need to change our operators so that they read this value instead of the parameter they currently receive using the `config` repository.
+
+##### List of controllers that reconcile clusters and use the `baseDomain`
```

Comment:
Just a side note, if we want our controllers on the MC to get the baseDomain value from the Cluster CR, we will introduce a dependency on cluster api to all of them. (e.g. dex-operator does not currently have this dependency)

Reply:
True. I think only dex-operator doesn't depend on CAPI already, so yeah, it would require one new dependency for that controller. I personally think it's not a blocker, but I'll add this as a disadvantage.